src/sagemaker/lineage/query.py

```diff
@@ -176,13 +228,209 @@ def _artifact_to_lineage_object(self):
         return Artifact.load(artifact_arn=self.arn, sagemaker_session=self._session)
 
 
+class PyvisVisualizer(object):
+    """Create object used for visualizing graph using Pyvis library."""
+
+    def __init__(self, graph_styles):
+        """Init for PyvisVisualizer.
+
+        Args:
+            graph_styles: A dictionary that contains graph style for node and edges by their type.
+                Example: Display the nodes with different color by their lineage entity / different
+                    shape by start arn.
+                        lineage_graph = {
+                            "TrialComponent": {
+                                "name": "Trial Component",
+                                "style": {"background-color": "#f6cf61"},
+                                "isShape": "False",
+                            },
+                            "Context": {
+                                "name": "Context",
+                                "style": {"background-color": "#ff9900"},
+                                "isShape": "False",
+                            },
+                            "StartArn": {
+                                "name": "StartArn",
+                                "style": {"shape": "star"},
+                                "isShape": "True",
+                                "symbol": "★",  # shape symbol for legend
+                            },
+                        }
+
+        """
+        # import visualization packages
+        (
+            self.Network,
+            self.Options,
+            self.IFrame,
+            self.BeautifulSoup,
+        ) = self._import_visual_modules()
+
+        self.graph_styles = graph_styles
+
+        # pyvis graph options
+        self._options = """
+            var options = {
+            "configure":{
+                "enabled": false
+            },
+            "layout": {
+                "hierarchical": {
+                    "enabled": true,
+                    "blockShifting": true,
+                    "direction": "LR",
+                    "sortMethod": "directed",
+                    "shakeTowards": "leaves"
+                }
+            },
+            "interaction": {
+                "multiselect": true,
+                "navigationButtons": true
+            },
+            "physics": {
+                "enabled": false,
+                "hierarchicalRepulsion": {
+                    "centralGravity": 0,
+                    "avoidOverlap": null
+                },
+                "minVelocity": 0.75,
+                "solver": "hierarchicalRepulsion"
+            }
+        }
+            """
+
+    def _import_visual_modules(self):
+        """Import modules needed for visualization."""
+        get_module("pyvis")
```

Comment:
this mechanism is inconsistent with modifying setup.py - in that case you can just explicitly declare pyvis in imports without needing `get_module`